### PR TITLE
PYTHON-1631 Document new release process

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

File: RELEASE.rst [hunk 1]
```diff
@@ -31,39 +31,74 @@ that changes the major version number.
...
+6. Commit with a BUMP version_number message, eg `git commit -m 'BUMP 3.11.0'`.
+
+7. Tag w/ version_number, eg, `git tag -a '3.11.0' -m '3.11.0' <COMMIT>`.
+
+8. Push commit / tag, eg `git push && git push --tags`.
+
+9. Pushing a tag will trigger a release process in Evergreen which builds
+   wheels and eggs for manylinux, macOS, and Windows. Wait for these jobs to
+   complete and then download the "Release files" archive from each task. See:
+   https://evergreen.mongodb.com/waterfall/mongo-python-driver?bv_filter=release
```
Comment (prashantmital) [inline RELEASE.rst:60]: Should this be a Spruce link instead to ensure it doesn't go stale soon? I am not sure if the evergreen URLs are going away but Spruce certainly seems to be the future.

File: RELEASE.rst [hunk 2]
```diff
@@ -31,39 +31,74 @@ that changes the major version number.
...
    them. You can also run the doctests: `python setup.py doc -t`.
 
-2. Add release notes to doc/changelog.rst. Generally just summarize/clarify
+2. Check Jira to ensure all the tickets in this version have been completed.
+
+3. Add release notes to doc/changelog.rst. Generally just summarize/clarify
    the git log, but you might add some more long form notes for big changes.
 
-3. Search and replace the "devN" version number w/ the new version number (see
+4. Search and replace the "devN" version number w/ the new version number (see
```
Comment (prashantmital) [inline RELEASE.rst:46]: What note is being referred to here?

File: RELEASE.rst [hunk 3]
```diff
@@ -31,39 +31,74 @@ that changes the major version number.
...
+     pymongo-<version>-cp27-cp27m-macosx_10_14_intel.whl
+     ...
+     pymongo-<version>-py2.7-macosx-10.14-intel.egg
+     $ ls path/to/windows/
+     pymongo-<version>-cp27-cp27m-win32.whl
+     ...
+     pymongo-<version>-cp38-cp38-win_amd64.whl
 
-7. Push commit / tag.
+10. Build the source distribution::
```
Comment (prashantmital) [inline RELEASE.rst:78]: Hmm, I wish we didn't have to do this manually. I think we should create a separate ticket to make creating the sdist part of the automated release on Evergreen. IIUC we could just build the source dist in any of the platform-specific builds prior to compiling the C-extensions.

File: RELEASE.rst (hunk 1 above)
Comment (ShaneHarvey) [reply RELEASE.rst:60]: Spruce doesn't have a waterfall page yet, it only supports viewing patch builds.

File: RELEASE.rst (hunk 2 above)
Comment (ShaneHarvey) [reply RELEASE.rst:46]: I assume it's the note in Versioning. I added a link.

File: RELEASE.rst (hunk 3 above)
Comment (ShaneHarvey) [reply RELEASE.rst:78]: I'll add this to the "sign releases" epic.

**Review by prashantmital — APPROVED** [review]
Looks like its just the change stream tests failing which can be handled separately. LGTM!

**Pull request merged**